python/paddle/fluid/contrib/int8_inference/README.md

```diff
@@ -0,0 +1,75 @@
+
+## 0. Prerequisite
+You need to install at least PaddlePaddle-1.3 python package `pip install paddlepaddle==1.3`.
+
+## 1. How to generate INT8 model
+You can refer to the unit test in [test_calibration.py](../tests/test_calibration.py). Basically, there are three steps:
+* Construct calibration object.
+
+```python
+calibrator = int8_utility.Calibrator( # Step 1
+    program=infer_program, # required, FP32 program
+    pretrained_model=model_path, # required, FP32 pretrained model
+    algo=algo, # required, calibration algorithm; default is max, the alternative is KL (Kullback–Leibler divergence)
+    exe=exe, # required, executor
+    output=int8_model, # required, INT8 model
+    feed_var_names=feed_dict, # required, feed dict
+    fetch_list=fetch_targets) # required, fetch targets
+```
+
+* Call the calibrator.sample_data() after executor run.
+```python
+_, acc1, _ = exe.run(
+    program,
+    feed={feed_dict[0]: image,
+          feed_dict[1]: label},
+    fetch_list=fetch_targets)
+
+calibrator.sample_data() # Step 2
+```
+
+* Call the calibrator.save_int8_model() after sampling over specified iterations (e.g., iterations = 50)
+```python
+calibrator.save_int8_model() # Step 3
+```
+
+## 2. How to run INT8 model
+You can load INT8 model by load_inference_model [API](https://github.com/PaddlePaddle/Paddle/blob/8b50ad80ff6934512d3959947ac1e71ea3fb9ea3/python/paddle/fluid/io.py#L991) and run INT8 inference similar as [FP32](https://github.com/PaddlePaddle/models/blob/develop/fluid/PaddleCV/object_detection/eval.py "FP32").
+
+```python
+[infer_program, feed_dict,
+    fetch_targets] = fluid.io.load_inference_model(model_path, exe)
+```
+
+## 3. Result
+We provide the results of accuracy and performance, measurd on [Intel® Xeon® Platinum Gold Processor](https://ark.intel.com/products/120489/Intel-Xeon-Gold-6148-Processor-27-5M-Cache-2-40-GHz- "Intel® Xeon® Gold 6148 Processor") (also known as Intel® Xeon® Skylake6148).
+
+| Model  | Dataset  | FP32 Accuracy  | INT8 Accuracy  | Accuracy Diff  |
+| ------------ | ------------ | ------------ | ------------ | ------------ |
+| ResNet-50  | Small  | 72.00%  | 72.00%  |  0.00% |
+| MobileNet-V1  | Small  | 62.00%  | 62.00%  | 0.00%  |
+| ResNet-50  | Full ImageNet Val  |  76.63%  | 76.14%  | 0.48% |
+| MobileNet-V1 | Full ImageNet Val  | 70.78%  | 70.41%  | 0.37%  |
+
+Please note that [Small](http://paddle-inference-dist.cdn.bcebos.com/int8/calibration_test_data.tar.gz "Small") is a subset of [full ImageNet validation dataset](http://www.image-net.org/challenges/LSVRC/2012/nnoupb/ILSVRC2012_img_val.tar "full ImageNet validation dataset"). 
+
+| Model  | Batch Size  | FP32 Throughput (Images/second)  | INT8 Throughput (Images/second)  | Improvement  |
+| ------------ | ------------ | ------------ | ------------ | ------------ |
+| ResNet-50  | 1  |   14  | 23  | 1.59X |
+| MobileNet-V1 | 1  | 60  | 97  | 1.61X  |
```

Comment:
This is my result on Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (small dataset).
```
Start INT8 inference for MobileNet-V1 on 50 images ...
FP32 MobileNet-V1: throughput 28.0962518271 images/second, latency 0.0355919361115 second, accuracy 0.62
INT8 MobileNet-V1: throughput 2.44348370827 images/second, latency 0.409251756668 second, accuracy 0.62
```
```
Start INT8 inference for ResNet-50 on 50 images ...
FP32 ResNet-50: throughput 7.22583899537 images/second, latency 0.138392233849 second, accuracy 0.72
INT8 ResNet-50: throughput 1.20720167195 images/second, latency 0.828362007141 second, accuracy 0.72
```
Besides, could you add `batchsize=xxx` in the log?

Reply:
Below is the test on a virtual machine, after I change to a physical machine:
```
FP32 MobileNet-V1: batch_size 1 throughput 45.1946017808 images/second, latency 0.0221265363693 second, accuracy 0.62
INT8 MobileNet-V1: batch_size 1 throughput 68.8913307349 images/second, latency 0.0145156145096 second, accuracy 0.62

FP32 ResNet-50: batch_size 1 throughput 10.5143084503 images/second, latency 0.0951084899902 second, accuracy 0.72
INT8 ResNet-50: batch_size 1 throughput 14.4883329359 images/second, latency 0.0690210533142 second, accuracy 0.72
```